solcore/solar_cell_solver.py

```diff
@@ -279,3 +289,18 @@ def prepare_solar_cell(solar_cell, options):
 
     if options.position is None:
```

Comment:
Maybe now that check is much more complex with multiple conditionals, it will be convenient to put it in a separate function that takes whatever inputs needed and returns the value to be assigned to `options.positions`. It will make it easier to test the functionality on its own.